src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptor.java

```diff
@@ -0,0 +1,435 @@
+     * controlled by this descriptor (which can happen if they have different qualified names
+     * within the base package used by the descriptor) since the user has no way to disambiguate
+     * these on the command line).
+     *
+     * @param pluggableClass a plugin class discovered by the command line parser that
+     *                       was not rejected by {@link #getClassFilter}
+     * @return the instantiated object that will be used by the command line parser
+     * as an argument source
+     * @throws IllegalAccessException
+     * @throws InstantiationException
+     */
+    @Override
+    public Object getInstance(Class<?> pluggableClass) throws IllegalAccessException, InstantiationException {
+        Annotation annot = null;
+        final String simpleName = pluggableClass.getSimpleName();
+
+        if (allDiscoveredAnnotations.containsKey(simpleName)) {
+            // we found a plugin class with a name that collides with an existing class;
+            // plugin names must be unique even across packages
+            throw new IllegalArgumentException(
+                    String.format("A plugin class name collision was detected (%s/%s). " +
+                                    "Simple names of plugin classes must be unique across packages.",
+                            pluggableClass.getName(),
+                            allDiscoveredAnnotations.get(simpleName).getClass().getName())
+            );
+        } else if (toolDefaultAnnotations.containsKey(simpleName)) {
+            // an instance of this class was provided by the tool as one of it's default annotations;
+            // use the default instance as the target for command line argument values
+            // rather than creating a new one, in case it has state provided by the tool
+            annot = toolDefaultAnnotations.get(simpleName);
+        } else {
+            annot = (Annotation) pluggableClass.newInstance();
+        }
+
+        // Add all annotations to the allDiscoveredAnnotations list, even if the instance came from the
+        // tool defaults list (we want the actual instances to be shared to preserve state)
+        allDiscoveredAnnotations.put(simpleName, annot);
+
+        // Dynamic discovery of annotation groups
+        Class<?>[] interfaces = annot.getClass().getInterfaces();
+        for (Class<?> inter : interfaces) {
+            // Following with how groups are currently defined and discovered, namely they are interfaces that
+            // extend Annotation, groups are discovered by interrogating annotations for their interfaces and
+            // associating the discovered annotations with their defined groups.
+            if ((inter != pluginBaseClass) && (pluginBaseClass.isAssignableFrom(inter))) {
+                discoveredGroups.merge(inter.getSimpleName(), Collections.singletonList(annot), (a, b) -> {return Stream.concat(a.stream(),b.stream()).collect(Collectors.toList());} );
```

Comment:
Looks good. The "return" and brackets in the lambda are unnecessary and can be removed:

`(a, b) -> Stream.concat(a.stream(),b.stream()).collect(Collectors.toList()) `

Reply:
done